docs/spec/fastsync/fastsync.tla

```diff
@@ -530,40 +574,59 @@ SendStatusResponseMessage(pState) ==
 
 SendAddPeerMessage ==
    \E peer \in AllPeerIds:
-     /\ inMsg' = [type |-> "addPeer", peerId |-> peer]
-     /\ UNCHANGED peersState
+     inMsg' = AsInMsg([type |-> "addPeer", peerId |-> peer])
 
 SendRemovePeerMessage ==
    \E peer \in AllPeerIds:
-     /\ inMsg' = [type |-> "removePeer", peerId |-> peer]
-     /\ UNCHANGED peersState
+     inMsg' = AsInMsg([type |-> "removePeer", peerId |-> peer])
 
 SendSyncTimeoutMessage ==
-    /\ inMsg' = [type |-> "syncTimeout"]
-    /\ UNCHANGED peersState
+     inMsg' = AsInMsg([type |-> "syncTimeout"])
 
 
 SendControlMessage ==
     \/ SendAddPeerMessage
     \/ SendRemovePeerMessage
     \/ SendSyncTimeoutMessage
 
+\* An extremely important property of block hashes (blockId):
+\* If a commit is correct, then the previous block in the block store must be also correct.
+UnforgeableBlockId(height, block) ==
+    block.hashEqRef => block = chain[height]
```

Comment:
We should improve the explanation. It talks about commit mit commit does not show up in formula.